### DISCUSSION TITLE
[Question] How can we differentiate between Forge and NeoForge mods?

### DISCUSSION THREAD

**Discussion by Kira-NT** [neoforged/NeoForge#58] [Q&A] [answered]
Hi folks! I'm a maintainer of a GitHub Action that aids users in publishing their Minecraft projects across various platforms, such as Modrinth and CurseForge. At present, my action identifies NeoForge mods as Forge mods, which is in line with what was stated on your website:

> At the present time, mods built by either system should be intercompatible between forks.

However, when the transition to become a hard fork takes place, how will we be able to distinguish between Forge and NeoForge mods? Is it possible that the `forge` dependency in `mods.toml` will become `neoforge`?

```diff
[[dependencies.foo]]
-    modId="forge"
+    modId="neoforge"
```

Alternatively, could there be plans to entirely revamp the metadata format and adopt a naming style similar to Fabric and Quilt, such as `neoforge.mod.(json|toml)`?

**Comment by shartte** [comment]
I think this will only resolve itself past 1.20.1
I believe for now you can rely on people specifying it manually in your GH Action (or, you default to [forge, neoforge] if you detect MC version 1.20.1)

**Comment by DereC4** [comment]
wait are forge mods compatible with neoforge?

**Reply by Kira-NT** [reply]
NeoForge is a fork of Forge. Thus, there was a brief period when NeoForge and Forge were essentially the same, distinguished only by their branding. However, as the time goes, Forge continues to stagnate, while NeoForge continues to evolve with new features and clean up the issues inherited from Forge. Consequently, they grow apart more and more every day, and at present, Forge mods are generally not compatible as is with NeoForge anymore.

**Comment by Kira-NT** [comment, marked as answer]
Answering my own question, my assumption was correct – NeoForge now utilizes the `neoforge` dependency instead of the legacy `forge` one:

```diff
[[dependencies.foo]]
-    modId="forge"
+    modId="neoforge"
```
